Trying to link with SDL3 static built with SDL_JOYSTICK=OFF gives undefined reference to SDL_VIRTUAL_JoystickDriver

If you build SDL3 with these commands:

`    git clone https://github.com/libsdl-org/SDL.git
    cd SDL/
    git checkout release-3.2.x
    cmake -DSDL_STATIC=ON -DSDL_SHARED=OFF -DCMAKE_BUILD_TYPE=Release       -DSDL_AUDIO=OFF   -DSDL_VIDEO=ON   -DSDL_GPU=ON   -DSDL_RENDER=ON   -DSDL_CAMERA=OFF   -DSDL_JOYSTICK=OFF   -DSDL_HAPTIC=OFF   -DSDL_HIDAPI=OFF   -DSDL_POWER=OFF   -DSDL_SENSOR=OFF   -=OFF -DSDL_DIALOG=OFF       -B build .
    cmake --build build --config Release --parallel`

And then try to link a minimal application (ie, create a window and fill white), you'll receive:

[build] /usr/bin/ld: /usr/local/lib/libSDL3.a(SDL_joystick.c.o): in function SDL_OpenJoystick_REAL':
[build] SDL_joystick.c:(.text+0x7a5a): undefined reference to SDL_VIRTUAL_JoystickDriver'

@slouken,

The error is on release-3.2.x branch, tested with the latest commit `01ef4c46a1731400bb9760434c4e09329540b9b7`.

I can reproduce by using the copy button, pasting this as is on the terminal, waiting until everything fishes up and pressing `ENTER`. If you don't see the linker error, probably it is a missing dependency on my machine which the configure does not warn me about. But, if I disabled Josystick with `-DSDL_JOYSTICK=OFF`, why am I receiving this linker error?

I get:
/usr/bin/ld: ./build//libSDL3.a(SDL_joystick.c.o): in function `SDL_OpenJoystick_REAL':
SDL_joystick.c:(.text+0x7a5a): undefined reference to `SDL_VIRTUAL_JoystickDriver'
collect2: error: ld returned 1 exit status

Using gcc and a .c file gives the same error:
gcc minimal.c -L./build/ -lSDL3 -ldl -pthread -lm

```bash
git clone https://github.com/libsdl-org/SDL.git
cd SDL/
git checkout release-3.2.x
cmake -DSDL_STATIC=ON -DSDL_SHARED=OFF -DCMAKE_BUILD_TYPE=Release -DSDL_AUDIO=OFF -DSDL_VIDEO=ON -DSDL_GPU=ON -DSDL_RENDER=ON -DSDL_CAMERA=OFF -DSDL_JOYSTICK=OFF -DSDL_HAPTIC=OFF -DSDL_HIDAPI=OFF -DSDL_POWER=OFF -DSDL_SENSOR=OFF -=OFF -DSDL_DIALOG=OFF -B build .
cmake --build build --config Release --parallel
echo -e "#include <SDL3/SDL.h> \n int main() { SDL_Init(SDL_INIT_VIDEO); return 0; }" > minimal.cpp
g++ minimal.cpp -L./build/ -lSDL3 -ldl -pthread
```

Sorry for the previous comments, I was busy and could not give much details.

--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1142,7 +1142,11 @@ SDL_Joystick *SDL_OpenJoystick(SDL_JoystickID instance_id)
     joystick->attached = true;
     joystick->led_expiration = SDL_GetTicks();
     joystick->battery_percent = -1;
+#ifdef SDL_JOYSTICK_VIRTUAL
     joystick->is_virtual = (driver == &SDL_VIRTUAL_JoystickDriver);
+#else
+    joystick->is_virtual = false;
+#endif
 
     if (!driver->Open(joystick, device_index)) {
         SDL_SetObjectValid(joystick, SDL_OBJECT_TYPE_JOYSTICK, false);

